src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SignatureVerificationCache.cs

```diff
@@ -115,15 +117,17 @@ private void ValidateStringArgumentNotNullOrEmpty(string stringArgValue, string
         private void TrimCacheIfNeeded()
         {
             // If the size of the cache exceeds the threshold, set that we are in trimming and trim the cache accordingly.
-            long currentCacheSize = _cache.GetCount();
+            long currentCacheSize = _cache.Count;
             if ((currentCacheSize > _cacheSize + _cacheTrimThreshold) && (0 == Interlocked.CompareExchange(ref _inTrim, 1, 0)))
             {
                 try
                 {
-                    _cache.Trim((int)(((double)(currentCacheSize - _cacheSize) / (double)currentCacheSize) * 100));
+                    // Example: 2301 - 2000 = 301; 301 / 2301 = 0.1308 * 100 = 13% compacting
+                    _cache.Compact((int)(((double)(currentCacheSize - _cacheSize) / (double)currentCacheSize) * 100));
```

Comment:
`MemoryCache.Compact` consumes a double so we don't need to cast with (int)

https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.caching.memory.memorycache.compact?view=net-8.0

Reply:
Casting to int was removed as suggested.